core/common/src/serializers/DateTimeUnitSerializers.kt

```diff
@@ -98,12 +96,12 @@ public object DayBasedDateTimeUnitSerializer: KSerializer<DateTimeUnit.DayBased>
                             seen = true
                         }
                         CompositeDecoder.DECODE_DONE -> break@loop // https://youtrack.jetbrains.com/issue/KT-42262
-                        else -> throw UnknownFieldException(elementIndex)
+                        else -> throwUnknownIndexException(elementIndex)
                     }
                 }
             }
         }
-        if (!seen) throw MissingFieldException("days")
+        if (!seen) throw MissingFieldException(missingField = "days", serialName = TimeBasedDateTimeUnitSerializer.descriptor.serialName)
```

Comment:
This looks suspicious. Is it intended to reference the descriptor of another type?

Reply:
No, it's not. Thanks!